client/oidc.go

```diff
@@ -1,3 +1,19 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
 package client
```

Comment:
I think this deserves its own package: `validator`. Helps with distinction architecturally and organizationally.

Reply:
I agree. I want to move validating into the a validator package, but I experimented with this in some draft PRs. The main issue I ran into is the cyclical dependencies between the client and providers that are exposed when client depends on the validator package. My next PR after this PR is fixing the cyclical dependencies in the client, which should enable creating a nice validator package.